romanesco/format/__init__.py

```diff
@@ -54,17 +74,113 @@ def csv_to_rows(input):
 
     return output
 
-converters = {}
-validators = {}
+
+def converter_path(source, target):
+    """Gives the shortest path that should be taken to go from a source
+    type/format to a target type/format.
+
+    Throws a ``NetworkXNoPath`` exception if it can not find a path.
+
+    :param source: Validator tuple indicating the type/format being converted `from`.
+    :param target: ``Validator`` tuple indicating the type/format being converted `to`.
+    :returns: An ordered list of the analyses that need to be run to convert from
+    ``source`` to ``target``.
+    """
+    def get_edge_analysis(source, target):
+        """Return the analysis associated with an edge in the conversion graph,
+        given the source Validator and the target Validator).
+        """
+        for (u, v, data) in conv_graph.edges([source], data=True):
+            if v == target:
+                return data
+
+        return {}
+
+    # These are to ensure an exception gets thrown if source/target don't exist
+    get_validator(source)
+    get_validator(target)
+
+    # We sort and pick the first of the shortest paths just to produce a stable
+    # conversion path. This is stable in regards to which plugins are loaded at the
+    # time.
+    paths = all_shortest_paths(conv_graph, source, target)
+    path = sorted(paths)[0]
+    path = zip(path[:-1], path[1:])
+
+    return [get_edge_analysis(u, v) for (u, v) in path]
+
+
+def has_converter(source, target=Validator(type=None, format=None)):
+    """Determines if any converters exist from a given type, and optionally format.
+
+    Underneath, this just traverses the edges until it finds one which matches the
+    arguments.
+
+    :param source: ``Validator`` tuple indicating the type/format being converted `from`.
+    :param target: ``Validator`` tuple indicating the type/format being converted `to`.
+    :returns: ``True`` if it can converter from ``source`` to ``target``, ``False``
+    otherwise.
+    """
+    for (u, v) in conv_graph.edges():
+        if source.type and source.type != u.type:
+            continue
+
+        if source.format and source.format != u.format:
+            continue
+
+        if target.type and target.type != v.type:
+            continue
+
+        if target.format and target.format != v.format:
+            continue
+
+        return True
+
+    return False
```

Comment:
This function seems to only look for direct connections rather than doing a full graph traversal. If that is intentional, we should document that fact, since otherwise someone might think this simply checks if the conversion is possible at all.